src/goto-harness/recursive_initialization.h

```diff
@@ -69,6 +81,21 @@ class recursive_initializationt
     std::size_t depth,
     const recursion_sett &known_tags,
     code_blockt &body);
+  void initialize_array(
+    const exprt &array,
+    std::size_t depth,
+    const recursion_sett &known_tags,
+    code_blockt &body);
+  void initialize_dynamic_array(
+    const exprt &pointer,
+    std::size_t depth,
+    const recursion_sett &known_tags,
+    code_blockt &body);
+
+  bool should_be_treated_as_array(const irep_idt &pointer_name) const;
+  bool is_array_size_parameter(const irep_idt &cmdline_arg) const;
+  optionalt<irep_idt>
+  get_associated_size_variable(const irep_idt &array_name) const;
```

Comment:
⛏ These three functions are wrappers around one-liners used only once, I think they should be inlined.